embed-code-go/fragmentation/fragmentation.go

```diff
@@ -0,0 +1,211 @@
+	fragmentation.SourcesRoot = absoluteSourcesRoot
+	if err != nil {
+		panic(err)
+	}
+
+	absoluteCodeFile, err := filepath.Abs(codeFileRelative)
+	fragmentation.CodeFile = absoluteCodeFile
+	if err != nil {
+		panic(err)
+	}
+
+	fragmentation.Configuration = config
+
+	return fragmentation
+}
+
+//
+// Private methods
+//
+
+func (fragmentation Fragmentation) parseLine(line string, contentToRender []string, fragmentBuilders map[string]*FragmentBuilder) ([]string, map[string]*FragmentBuilder, error) {
+	cursor := len(contentToRender)
+
+	fragmentStarts := GetFragmentStarts(line)
+	fragmentEnds := GetFragmentEnds(line)
+
+	if len(fragmentStarts) > 0 {
+		for _, fragmentName := range fragmentStarts {
+			fragment, exists := fragmentBuilders[fragmentName]
+			if !exists {
+				builder := FragmentBuilder{FileName: fragmentation.CodeFile, Name: fragmentName}
+				fragmentBuilders[fragmentName] = &builder
+				fragment = fragmentBuilders[fragmentName]
+			}
+			fragment.AddStartPosition(cursor)
+		}
+	} else if len(fragmentEnds) > 0 {
+		for _, fragmentName := range fragmentEnds {
+			if fragment, exists := fragmentBuilders[fragmentName]; exists {
+				fragment.AddEndPosition(cursor - 1)
+			} else {
+				return nil, nil, fmt.Errorf("Cannot end the fragment `%s` as it wasn't started.", fragmentName)
+			}
+		}
+	} else {
+		contentToRender = append(contentToRender, line)
+	}
+	return contentToRender, fragmentBuilders, nil
+}
+
+func (fragmentation Fragmentation) targetDirectory() string {
+	fragmentsDir := fragmentation.Configuration.FragmentsDir
+	codeRoot, err := filepath.Abs(fragmentation.Configuration.CodeRoot)
+	if err != nil {
+		panic(fmt.Sprintf("Error calculating absolute path: %v", err))
+	}
+	relativeFile, err := filepath.Rel(codeRoot, fragmentation.CodeFile)
+	if err != nil {
+		panic(fmt.Sprintf("Error calculating relative path: %v", err))
+	}
+	subTree := filepath.Dir(relativeFile)
+	return filepath.Join(fragmentsDir, subTree)
+}
+
+//
+// Public methods
+//
+
+// Splits the file into fragments.
+// @return (content, fragments) a refined content of the file to be cut into fragments, and the Fragments
+func (fragmentation Fragmentation) Fragmentize() ([]string, map[string]Fragment, error) {
+	fragmentBuilders := make(map[string]*FragmentBuilder)
+	var contentToRender []string
+
+	file, err := os.Open(fragmentation.CodeFile)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text() + "\n"
+		contentToRender, fragmentBuilders, err = fragmentation.parseLine(line, contentToRender, fragmentBuilders)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	fragments := make(map[string]Fragment)
+	for k, v := range fragmentBuilders {
+		fragments[k] = v.Build()
+	}
+	fragments[DefaultFragment] = CreateDefaultFragment()
+
+	return contentToRender, fragments, nil
+
+}
+
+// WriteFragments serializes fragments to the output directory.
```

Comment:
`WriteFragments` is redunant, as you are documenting this very method. People are not blind.
Please start with "Serializes".